dataplicity/m2m/packetbase.py

```diff
@@ -31,6 +31,9 @@ def __new__(mcs, name, bases, attrs):
             if packet_cls.type >= 0:
                 assert packet_cls.type not in packet_cls.registry, "packet type {!r} has already been registered".format(packet_cls, type)
                 packet_cls.registry[packet_cls.type] = packet_cls
+                # register the packet type in base class so we can have a
+                # generic interface of accessing it.
+                bases[0].registry[packet_cls.type] = packet_cls
```

Comment:
Hmm I am doing exactly that in the test code, but I thought that it has started to work only after I added the line in question. Will test this and revert the line in case I was totally confused